src/github/client/github-client.ts

```diff
@@ -31,15 +32,32 @@ export default class GitHubClient {
 	private readonly githubInstallationId: InstallationId;
 	private readonly logger: Logger;
 
-	constructor(
+	public static async build(githubInstallationId: InstallationId,
+		logger: Logger,
+		appTokenHolder: AppTokenHolder = AppTokenHolder.getInstance()) : Promise<GitHubClient> {
+		const timeout = await numberFlag(NumberFlags.GITHUB_CLIENT_TIMEOUT, 30000);
+		return new GitHubClient(githubInstallationId, logger, appTokenHolder, timeout);
```

Comment:
I had to make a builder method here, and made the constructor private.

Because constructor can't be async